### PR TITLE
Remove `verbose_generic_activity_with_arg`

### PR DESCRIPTION
**Pull request by Zoxc**
This removes `verbose_generic_activity_with_arg` and changes users to `generic_activity_with_arg`. This keeps the output of `-Z time` readable while these repeated events are still available with the self profiling mechanism.

### CODE REVIEW

File: compiler/rustc_mir_transform/src/pass_manager.rs [hunk 1]
```diff
@@ -121,7 +121,10 @@ fn run_passes_inner<'tcx>(
                 validate_body(tcx, body, format!("before pass {name}"));
             }
 
-            tcx.sess.time(name, || pass.run_pass(tcx, body));
+            tcx.sess
+                .prof
+                .generic_activity_with_arg("mir_pass", name)
```
Comment (cjgillot) [inline compiler/rustc_mir_transform/src/pass_manager.rs:126]: Could you leave the pass name as activity name? `body.source.def_id()` can be used as arg.

Comment (Zoxc) [reply compiler/rustc_mir_transform/src/pass_manager.rs:126]: That should work, but they should gain a `mir_` prefix to fit with the profiling naming scheme.

**Review by wesleywiser — APPROVED** [review]
Seems reasonable, no objections. 

r=me with build passing

**Pull request merged**